### PR TITLE
[REGEDIT] Handle Ctrl+Backspace in AddressBar

### PR DESCRIPTION
**Pull request by whindsaks**
Handle <kbd>Ctrl</kbd><kbd>Backspace</kbd> in the AddressBar like Windows 10 by deleting spans of special and/or normal characters.

Notes:
- Also fixes a bug where <kbd>Del</kbd> does not work in the AddressBar nor during Tree/List label edit because it gets eaten by the accelerator.
- Removed the code that handles <kbd>Ctrl</kbd><kbd>A</kbd> in the AddressBar because the Edit control does this out of the box.

### CODE REVIEW

**Review by katahiromz — CHANGES_REQUESTED** [review]
https://github.com/reactos/reactos/blob/4028de8c55c7f590a6a04c6ae120012e158d6e3f/dll/win32/browseui/CAutoComplete.cpp#L868

I think `Ctrl+BackSpace` should be implemented as AutoComplete.

See also: https://learn.microsoft.com/en-us/windows/win32/api/shldisp/nf-shldisp-iautocomplete-init

**Comment by katahiromz** [comment]
Steps:

1. Create the instance of `CLSID_AutoComplete` class.
2. Create an AutoComplete list of registry key pathes.
3. Pass `hwndEdit` and thmdAC list via `IAutoComplete::Init` method.

**Comment by whindsaks** [comment]
> I think `Ctrl+BackSpace` should be implemented as AutoComplete.

I suppose but I'm not dealing with providing the suggestion strings in this PR...

**Review by katahiromz — APPROVED** [review]
Currently `Ctrl+BackSpace` is working. Thank you.
https://github.com/user-attachments/assets/b46935d7-ce3b-4b9d-a34c-92315d5e6426

File: base/applications/regedit/main.c [hunk 1]
```diff
@@ -176,25 +176,36 @@ void ExitInstance(HINSTANCE hInstance)
...
 }
 
-BOOL TranslateChildTabMessage(PMSG msg)
+static BOOL InLabelEdit(HWND hWnd, UINT Msg)
 {
-    if (msg->message != WM_KEYDOWN) return FALSE;
-
-    /* Allow Ctrl+A on address bar */
-    if ((msg->hwnd == g_pChildWnd->hAddressBarWnd) &&
-        (msg->message == WM_KEYDOWN) &&
```
Comment (HBelusca) [inline base/applications/regedit/main.c:185]: Is the Ctrl-A now handled via the autocomplete thing?

**Review by HBelusca — COMMENTED** [review]
We should also note this:
https://jira.reactos.org/browse/CORE-1419
https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/user/user32/controls/edit.c;hb=4093d0e16444ac272c554c5fc36e9ad792b6fa08#l3685

By the way, the ctrl-backspace is kind of equivalent to: Ctrl-Shift-Left, followed by backspace.

File: base/applications/regedit/childwnd.c [hunk 2]
```diff
@@ -363,6 +423,12 @@ LRESULT CALLBACK ChildWndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lPa
                                                     IMAGE_ICON, 12, 12, 0);
         SendMessageW(g_pChildWnd->hAddressBtnWnd, BM_SETIMAGE, IMAGE_ICON, (LPARAM)g_pChildWnd->hArrowIcon);
 
+        if (SUCCEEDED(CoCreateInstance(&CLSID_AutoComplete, NULL, CLSCTX_INPROC_SERVER, &IID_IAutoComplete, (void**)&pAutoComplete)))
+        {
+            pAutoComplete->lpVtbl->Init(pAutoComplete, g_pChildWnd->hAddressBarWnd, (IUnknown*)&g_DummyEnumStrings, NULL, NULL);
+            pAutoComplete->lpVtbl->Release(pAutoComplete);
```
Comment (HBelusca) [inline base/applications/regedit/childwnd.c:429]: I wonder whether you could use the slightly more "elegant" helper macros (that should be in the autogenerated headers)
```suggestion
            IAutoComplete_Init(pAutoComplete, g_pChildWnd->hAddressBarWnd, (IUnknown*)&g_DummyEnumStrings, NULL, NULL);
            IAutoComplete_Release(pAutoComplete);
```

File: base/applications/regedit/main.c (hunk 1 above)
Comment (whindsaks) [reply base/applications/regedit/main.c:185]: Ctrl+A is handled by ComCtl32 v6 Edit which I assume is supported everywhere?

Comment (katahiromz) [reply base/applications/regedit/main.c:185]: Our regedit has a correct manifest for comctl32 v6.

File: base/applications/regedit/main.c [hunk 3]
```diff
@@ -205,6 +216,8 @@ int WINAPI wWinMain(HINSTANCE hInstance,
 
     UNREFERENCED_PARAMETER(hPrevInstance);
 
+    OleInitialize(0);
```
Comment (HBelusca) [inline base/applications/regedit/main.c:219]:
```suggestion
    OleInitialize(NULL);
```
(parameter is pointer-type)

**Review by HBelusca — APPROVED** [review]

**Pull request merged**